src/main/java/hudson/plugins/performance/TaurusParser.java

```diff
@@ -0,0 +1,163 @@
+package hudson.plugins.performance;
+
+import hudson.Extension;
+import org.apache.commons.io.FilenameUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parser for Taurus
+ */
+public class TaurusParser extends AbstractParser {
+
+    @Extension
+    public static class DescriptorImpl extends PerformanceReportParserDescriptor {
+        @Override
+        public String getDisplayName() {
+            return "Taurus";
+        }
+    }
+
+    @DataBoundConstructor
+    public TaurusParser(String glob) {
+        super(glob);
+    }
+
+    @Override
+    public String getDefaultGlobPattern() {
+        return "**/*.xml";
+    }
+
+    @Override
+    PerformanceReport parse(File reportFile) throws Exception {
+        return isXMLFileType(reportFile) ? readFromXML(reportFile) : readFromCSV(reportFile);
+    }
+
+    private boolean isXMLFileType(File reportFile) {
+        return FilenameUtils.getExtension(reportFile.getName()).toLowerCase().contains("xml");
```

Comment:
XML file criteria is not extension. Extension can be any. Actual file content is criteria.